kore/test/Test/Kore/Step/Axiom/Matcher.hs

```diff
@@ -969,6 +970,19 @@ test_matching_Map =
         ]
     ]
 
+test_matcherOverloading :: [TestTree]
+test_matcherOverloading =
```

Comment:
Can we add some tests for constructors that are overloaded, but don't match? I would also like to see some unit tests of `Kore.Step.Simplification.Overloading` directly.